packages/next/server/next-typescript.ts

```diff
@@ -196,16 +196,20 @@ export function createTSPlugin(modules: {
   }
 
   function create(info: ts.server.PluginCreateInfo) {
-    const appDir = path.join(info.project.getCurrentDirectory(), 'app')
+    const projectDir = info.project.getCurrentDirectory()
+    const appDir = new RegExp(
+      '^' + (projectDir + '(/src)?/app').replace(/[\\/]/g, '[\\/]')
```

Comment:
What’s `)?` here for? Will it have `src)` in the path for some case?

Reply:
I believe that's just a regex group, eg: `(/src)?`, as some projects maybe or not use the `src` folder to house their app.